tests/compile-fail/concurrency/unwind_top_of_stack.rs

```diff
@@ -1,18 +1,15 @@
 // ignore-windows: Concurrency on Windows is not supported yet.
-// error-pattern: calling a function with ABI C-unwind using caller ABI C
+// error-pattern: unwinding past the topmost frame of the stack
 
 //! Unwinding past the top frame of a stack is Undefined Behavior.
-//! However, it is impossible to do that in pure Rust since one cannot write an unwinding
-//! function with `C` ABI... so let's instead test that we are indeed correctly checking
-//! the callee ABI in `pthread_create`.
```

Comment:
This is possible again probably since rust-lang/rust#84158. :upside_down_face:
(Testing this since I added unwind past `extern "C"` detection.)

Reply:
This is a rustc bug though that we are deliberately leaving open... not sure how I feel about testing it.^^
At least, the test should explicitly says this.

Reply:
Actually all `compile-fail` tests of the unwinding ABI check relies on this bug... :upside_down_face:

Reply:
> Actually all compile-fail tests of the unwinding ABI check rely on this bug... 

I don't think that's true -- the Miri test suite has been passing before https://github.com/rust-lang/rust/pull/84158 landed.

The rustc bug will be fixed soon-ish. And then we'll have to revert this change to `unwind_top_of_stack.rs` again. So I don't see the point of your change here.

Reply:
> > Actually all compile-fail tests of the unwinding ABI check rely on this bug...
> 
> I don't think that's true -- the Miri test suite has been passing before [rust-lang/rust#84158](https://github.com/rust-lang/rust/pull/84158) landed.

Oh I mean the tests added this this PR, not the existing tests.

Reply:
Ah I see. That's not a good situation though, it means we'll lose all those tests when `c_unwind` is stabilized. We need at least a plan for what to do (and ideally comments in the test files explaining that plan).

Reply:
Either way, that's IMO not an argument to change this test that already *is* future-proved against `c_unwind`.